### ISSUE TITLE
pytorch warning in `dist_reduce`

### ISSUE THREAD

**Issue by ZhaofengWu** [allenai/allennlp#5568]
https://github.com/allenai/allennlp/blob/6614077be53f15fc2a3aee9a1b29447a72053720/allennlp/nn/util.py#L2150

`UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).`

**Comment by dirkgr** [comment]
That's a dumb warning. If you pass in a `List` it does the right thing. If you pass in a tensor, it also does the right thing, but complains about it.
